### PR TITLE
Remove some remaining CHERI MIPS references

### PR DESCRIPTION
**Pull request by bsdjhb**
- cheribsdbox: Remove MIPS references and workarounds.
- README.md: Swap out reference to MIPS for ARMv8-A (Morello)
- pci_user.c: Replace a MIPS reference with AArch64.
- KDB_WHY_CHERI: Remove MIPS reference.
- WITH_CHERI: Update list of relevant architectures.
- makeman: Workaround bsd.compiler.mk pickiness in CheriBSD.
- make targets: Add the aarch64c and riscv64c architectures.
- Move the CHERI_CAPREVOKE / CHERI dependency to the right file.
- Broken CHERI should implicitly disable CHERI_CAPREVOKE.
- Map aarch64c to the aarch64 LLVM target.
- Use more active language in CHERI build options to match upstream.
- src.opts.mk: Remove stale MK_(BINUTILS|GCC)_BOOTSTRAP references.
- Add build option descriptions for LIB64 and LIB64C.
- Add a build options file for WITHOUT_CHERIBSDBOX.
- src.conf.5: Regen.
- .mergify_pause_paths: Add src.conf.5.

### CODE REVIEW

**Comment by bsdjhb** [comment]
This started as a small PR to just trim some more MIPS references after I noticed some in cheribsdbox's Makefile.  I ended up down a rabbit hole of fixing the MIPS reference in WITH_CHERI that led to some build system fixes to get `make makeman` working and now using a fully regen'd src.conf.5.

File: share/mk/src.opts.mk [hunk 1]
```diff
@@ -432,10 +432,8 @@ BROKEN_OPTIONS+= TESTS
 .if ${.MAKE.OS} != "FreeBSD"
```
Comment (bsdjhb) [inline share/mk/src.opts.mk:432]: This whole .if is a local diff in CheriBSD that upstream doesn't seem to need, so we can probably drop it?

Comment (jrtc27) [reply share/mk/src.opts.mk:432]: Yes. Probably it got upstreamed somewhere slightly different so its eventual removal didn't result in this block being removed.

File: tools/build/options/WITHOUT_CHERI [hunk 2]
```diff
@@ -1,2 +1,2 @@
 .\" $FreeBSD$
-Set to not build software requiring CHERI capability support.
+Do not build software requiring CHERI capability support.
```
Comment (brooksdavis) [inline tools/build/options/WITHOUT_CHERI:2]:
```suggestion
Do not build optional software requiring CHERI capability support.
```
This mostly controls little test bits like cheribsdtest. (I've got a patch I'll submit shortly to make it only do this).

**Review by arichardson — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 CheriBSD extends FreeBSD to implement memory protection and
 software compartmentalization features supported by the CHERI ISA.
-CheriBSD includes support for CHERI extensions to the MIPS and RISC-V
-architectures.
+CheriBSD includes support for CHERI extensions to the ARMv8-A (via
```
Comment (jrtc27) [inline README.md:5]:
```suggestion
CheriBSD includes support for CHERI extensions to the Armv8-A (via
```

**Review by arichardson — APPROVED** [review]

File: tools/cheribsdbox/Makefile [hunk 4]
```diff
@@ -295,15 +287,9 @@ CRUNCH_ALIAS_w=	uptime
 CRUNCH_PROGS_usr.bin+=su
 
 # Tracing currently only implemented for CHERI-QEMU architectures
-.if ${MACHINE} == "mips" || ${MACHINE} == "riscv"
+.if ${MACHINE} == "riscv"
 CRUNCH_PROGS_usr.bin+=	qtrace
-.endif
-.if ${MACHINE_ARCH:Mmips64*} || ${MACHINE_ARCH:Mriscv64*}
 CRUNCH_PROGS_usr.bin+=	minimal_count_stats
```
Comment (jrtc27) [inline tools/cheribsdbox/Makefile:292]: This is separate, probably should be another condition

File: tools/cheribsdbox/Makefile [hunk 5]
```diff
@@ -295,15 +287,9 @@ CRUNCH_ALIAS_w=	uptime
 CRUNCH_PROGS_usr.bin+=su
 
 # Tracing currently only implemented for CHERI-QEMU architectures
```
Comment (jrtc27) [inline tools/cheribsdbox/Makefile:289]: Not quite true, CHERI-QEMU supports Morello but we don't have a magic tracing hint for it

File: tools/cheribsdbox/Makefile (hunk 4 above)
Comment (bsdjhb) [reply tools/cheribsdbox/Makefile:292]: Right now they would be duplicate, and we can always split it in the future if another architecture grows support for one of the other of these.

Comment (bsdjhb) [reply tools/cheribsdbox/Makefile:292]: I guess though the comment needs to only apply to qtrace

Comment (jrtc27) [reply tools/cheribsdbox/Makefile:292]: Yeah. Moving the comment would also work.

**Pull request merged**